### PR TITLE
fix pipeline cancelation

### PR DESCRIPTION
**Pull request by CheSema**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit 15e9009ccbe47ef057bee24d7129b4054e2e6205 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/72587/15e9009ccbe47ef057bee24d7129b4054e2e6205/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/71085/8aca23a4df9a39bd536c6cf8a2acc5ff8e71959a/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: tests/queries/0_stateless/03277_async_insert.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    INSERT INTO TABLE FUNCTION file('$FILE_CSV', 'CSV', 'c0 Int') SELECT c0 FROM t0;
+    INSERT INTO TABLE t0 (c0) FROM INFILE '$FILE_CSV' FORMAT CSV;
+"
+$CLICKHOUSE_CLIENT -q "SELECT * from t0"
+$CLICKHOUSE_CLIENT -q "DROP TABLE t0"
+
+$CLICKHOUSE_CLIENT -q "
+    SET async_insert = 1;
+    SELECT '${FILE_ARROW}';
+    CREATE TABLE t0 (c0 Int) ENGINE = Join(ANY, INNER, c0);
```
Comment (vdimir) [inline tests/queries/0_stateless/03277_async_insert.sh:33]: Is issue specific to `Join` table engine? If yes, what particularly cases it?

**Review by vdimir — APPROVED** [review]

File: tests/queries/0_stateless/03277_async_insert.sh (hunk 1 above)
Comment (CheSema) [reply tests/queries/0_stateless/03277_async_insert.sh:33]: No. It turns out that it related to the empty set of insertion data.

Comment (CheSema) [reply tests/queries/0_stateless/03277_async_insert.sh:33]: We created pipeline and destroyed it with no execution at all.

**Pull request merged**